lib/clientv2/ippool.go

```diff
@@ -88,6 +96,46 @@ func (r ipPools) Update(ctx context.Context, res *apiv2.IPPool, opts options.Set
 
 // Delete takes name of the IPPool and deletes it. Returns an error if one occurs.
 func (r ipPools) Delete(ctx context.Context, name string, opts options.DeleteOptions) (*apiv2.IPPool, error) {
+	// Deleting a pool requires a little care because of existing endpoints
+	// using IP addresses allocated in the pool.  We do the deletion in
+	// the following steps:
+	// -  disable the pool so no more IPs are assigned from it
+	// -  remove all affinities associated with the pool
+	// -  delete the pool
+	logCxt := log.WithField("Name", name)
+
+	// Get the pool so that we can find the CIDR associated with it.  If the pool is
+	// enabled we'll need to release affinities associated with the pool.
```

Comment:
Shouldn't we just do this even if the pool is enabled? 

Can a pool go from enabled to disabled while there are IPs allocated from it?

Reply:
hmmm - yeah actually now that you mention it that's a good idea!  it also closes down a window where failure just after disabling the pool means we wouldn't release affinities next attempt!